cmd/manager/operator.go

```diff
@@ -562,7 +561,7 @@ func createServiceMonitor(ctx context.Context, cfg *rest.Config, mClient *moncli
 			serviceMonitor.Spec.Endpoints[i].BearerTokenFile = serviceMonitorBearerTokenFile
 			serviceMonitor.Spec.Endpoints[i].TLSConfig = &monitoring.TLSConfig{
 				CAFile:     serviceMonitorTLSCAFile,
-				ServerName: serviceMonitorTLSServerName,
+				ServerName: "metrics." + namespace + ".svc",
```

Comment:
This the crux of the fix, right? The testing changes below aren't directly related to this change?

Reply:
Right, the testing changes are unrelated. (Our CI testing is a direct read from the metrics service, while the ServiceMonitor is so cluster monitoring can read the metrics)

Reply:
not atm. I can add it though

Reply:
@rhmdnd I refactored some to fit a simple unit test..